File: libs/opensource/sphinx/plugins/org.eclipse.sphinx.emf/src/org/eclipse/sphinx/emf/resource/OldResourceProviderAdapter.java
/**
 * <copyright>
 *
 * Copyright (c) 2015 itemis and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v2.0
 * which accompanies this distribution, and is available at
 * https://www.eclipse.org/org/documents/epl-2.0/EPL-2.0.html
 *
 * Contributors:
 *     itemis - Initial API and implementation
 *
 * </copyright>
 */
package org.eclipse.sphinx.emf.resource;

import org.eclipse.core.runtime.Assert;
import org.eclipse.emf.common.notify.Adapter;
import org.eclipse.emf.common.notify.Notifier;
import org.eclipse.emf.common.notify.impl.AdapterImpl;
import org.eclipse.emf.ecore.EObject;
import org.eclipse.emf.ecore.resource.Resource;

/**
 * An {@link Adapter adapter} for {@link EObject}s that implements {@link OldResourceProvider}.
 */
public class OldResourceProviderAdapter extends AdapterImpl implements OldResourceProvider {

	private Resource oldResource;

	public OldResourceProviderAdapter(Resource oldResource) {
		this.oldResource = oldResource;
	}

	/*
	 * @see org.eclipse.sphinx.emf.resource.OldResourceProvider#getOldResource()
	 */
	@Override
	public Resource getOldResource() {
		return oldResource;
	}

	/*
	 * @see org.eclipse.emf.common.notify.impl.AdapterImpl#isAdapterForType(java.lang.Object)
	 */
	@Override
	public boolean isAdapterForType(Object type) {
		return type == OldResourceProvider.class;
	}

	/*
	 * @see org.eclipse.emf.common.notify.impl.AdapterImpl#setTarget(org.eclipse.emf.common.notify.Notifier)
	 */
	@Override
	public void setTarget(Notifier newTarget) {
		Assert.isLegal(newTarget == null || newTarget instanceof EObject);
		super.setTarget(newTarget);
	}
}

File: cpp/emf-cpp/emf-sphinx/src/resource/OldResourceProviderAdapter.cpp
// OldResourceProviderAdapter.cpp
// 对齐 Java org.eclipse.sphinx.emf.resource.OldResourceProviderAdapter
#include "emf/sphinx/resource/OldResourceProviderAdapter.h"

namespace emf::sphinx::resource {

void OldResourceProviderAdapter::cacheOldContent(emf::common::Resource* /*res*/) {}
std::vector<emf::common::EObject*> OldResourceProviderAdapter::getOldContents(emf::common::Resource* /*res*/) { return {}; }
std::string OldResourceProviderAdapter::getOldURIFragment(emf::common::EObject* /*obj*/) { return ""; }
emf::common::EObject* OldResourceProviderAdapter::getOldEObject(const emf::common::URI& /*uri*/) { return nullptr; }
void OldResourceProviderAdapter::clearCache(emf::common::Resource* /*res*/) {}

}  // namespace emf::sphinx::resource
